### PR TITLE
fix: *matcher result should be boolean or number* for KeyGet2

### PR DESCRIPTION
**Pull request by Shivansh-yadav13**
Fix: https://github.com/casbin/node-casbin/issues/332

- added a case for result type string.
- please suggest changes if required.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@nodece @Zxilly plz review

**Comment by hsluoyz** [comment]
@Shivansh-yadav13 

1. Don't fix two bugs in one PR. Separate into two PRs.
2. AT the issue sender to review

**Comment by Shivansh-yadav13** [comment]
@tomfriedhof can you please review and suggest changes if required.

File: src/coreEnforcer.ts [hunk 1]
```diff
@@ -457,6 +457,13 @@ export class CoreEnforcer {
...
             break;
+          case 'string':
+            if (result === '') {
+              eftRes = Effect.Indeterminate;
+            } else {
+              eftRes = Effect.Allow;
+            }
+            break;
           default:
             throw new Error('matcher result should be boolean or number');
```
Comment (Zxilly) [inline src/coreEnforcer.ts:468]: please also update error hint

**Review by Zxilly — CHANGES_REQUESTED** [review]

**Comment by grzegorz-kap** [comment]
@Shivansh-yadav13 , when I was debugging it the value of 'result' was undefined.

In our implementation, we have multiple instances of Casbin enforcer.

Steps to reproduce:
1. Create a new Enforce with an empty permission set and call enforce
2. Create a new Enforce with some permissions and call enforce (an error is thrown: `matcher result should be boolean or number` - result is undefined)
3. Repeat step 2 (no error)



model:
```
[request_definition]
r = sub, dom, obj, act, v

[policy_definition]
p = sub, dom, obj, act, exp

[role_definition]
g = _, _, _

[policy_effect]
e = some(where (p.eft == allow))

[matchers]
m = g(r.sub, p.sub, r.dom) && r.dom == p.dom && r.obj == p.obj && r.act == p.act && eval(p.exp)
```

**Comment by Shivansh-yadav13** [comment]
@grzegorz-kap is it possible if you can show the code or any example because I'm not getting the same error by following the steps I might be doing something wrong.

Also, I realized that I don't get the error `matcher result should be boolean or number` when using `keyGet2` alone but when I use it with role_definition

**Comment by Shivansh-yadav13** [comment]
This is what happening

```js
const enforcer1 = await newEnforcer('examples/test_model.conf');
await enforcer1.enforce('sub', 'dom', '/data1', 'read', { office: 'london', floor: 'first' });

const enforcer2 = await newEnforcer('examples/test_model.conf', 'examples/test_policy.csv');
await enforcer2.enforce('sub', 'dom', '/data1', 'read', { office: 'london', floor: 'first' });
```
**Error -->** `matcher result should be boolean or number`

```js
const enforcer1 = await newEnforcer('examples/test_model.conf', 'examples/test_policy.csv');
await enforcer1.enforce('sub', 'dom', '/data1', 'read', { office: 'london', floor: 'first' });

const enforcer2 = await newEnforcer('examples/test_model.conf');
await enforcer2.enforce('sub', 'dom', '/data1', 'read', { office: 'london', floor: 'first' });
```
**No Error**

```js
const enforcer1 = await newEnforcer('examples/test_model.conf');
await enforcer1.enforce('sub', 'dom', '/data1', 'read', { office: 'london', floor: 'first' });

const enforcer2 = await newEnforcer('examples/test_model.conf');
await enforcer2.enforce('sub', 'dom', '/data1', 'read', { office: 'london', floor: 'first' });
```
**No Error**

**Comment by Shivansh-yadav13** [comment]
@grzegorz-kap Ig the error you are facing is resolved after #348, can you please confirm.

**Comment by nodece** [comment]
Could you add unit tests to cover this case?

**Review by Zxilly — APPROVED** [review]
lgtm

**Pull request merged**